crates/swc_css_minifier/src/compress/selector.rs

```diff
@@ -0,0 +1,267 @@
+use swc_common::DUMMY_SP;
+use swc_css_ast::*;
+use swc_css_visit::{VisitMut, VisitMutWith};
+
+pub fn compress_selector() -> impl VisitMut {
+    CompressSelector {}
+}
+
+struct CompressSelector {}
+
+impl CompressSelector {}
+
+impl VisitMut for CompressSelector {
+    fn visit_mut_nth(&mut self, nth: &mut Nth) {
+        nth.visit_mut_children_with(self);
+
+        match &nth.nth {
+            // `2n+1` + `odd`
```

Comment:
this case could be simplified even more: `2n+1`, `2n-1`, `-2n+5` are all odd; you should check whether `(a == 2 || a == -2) && b % 2 == 1` for odds and with last clause `b % 2 == 0` for evens; this covers then the case with `b == 0` below too.

Reply:
`-2n+5` is not `odd`
- `0` -> `-2 * 0 + 5` -> `5`
- `1` -> `-2 * 1 + 5` -> `3`
- `2` -> `-2 * 2 + 5` -> `1`
- `3` -> `-2 * 3 + 5` -> `-1`

Reply:
Also `2n+3` is not odd, but yes, I see some improvement

Reply:
well… -1 is also odd for me (as integer); that the very definition of oddness: yielding 1 from dividing (modulo) by 2.

this might be slightly different in various languages, not sure which is which in rust (see reminder vs. modulo operators, or various definition variants having different elements having same/different signs; cf. [wiki](https://en.wikipedia.org/wiki/Modulo_operation)).